pytorch_lightning/trainer/connectors/model_connector.py

```diff
@@ -35,6 +35,9 @@ def copy_trainer_model_properties(self, model):
         else:
             ref_model = model
 
+        automatic_optimization = ref_model.automatic_optimization and self.trainer.train_loop.automatic_optimization
+        self.trainer.train_loop.automatic_optimization = automatic_optimization
```

Comment:
should this be `or` for backwards compatibility? after this PR is merged, someone could set Trainer(automatic_optimization=False) but that would be overridden by the `automatic_optimization` set on the lightning module right?

Reply:
The current idea is that the default is False and you can turn it off by either the module or the trainer flag. So if one of them resolves to false, the result should be false as well.